tests/publiccloud/registercloudguest.pm

```diff
@@ -0,0 +1,62 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: cloud-regionsrv-client
+# Summary: Test system (re)registration
+# https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/integration_test-process.txt
+#
+# Maintainer: <qa-c@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use version_utils;
+use registration;
+use warnings;
+use testapi;
+use strict;
+use utils;
+use publiccloud::utils;
+
+sub run {
+    my ($self, $args) = @_;
+
+    # Preserve args for post_fail_hook
+    $self->{instance} = $args->{my_instance};
+
+    # Create $instance to make the code easier to read
+    my $instance = $args->{my_instance};
+    my $regcode = get_required_var('SCC_REGCODE') if (is_byos());
+    my $regcode_param = ($regcode) ? "-r $regcode" : '';
+
+    select_host_console();    # select console on the host, not the PC instance
+
+    # Test re-registration. Assuming the system has been registered before
+    $instance->run_ssh_command(cmd => '! sudo SUSEConnect -d', quiet => 0) if (is_byos());
+    $instance->run_ssh_command(cmd => 'sudo registercloudguest --clean', quiet => 0);
+    $instance->run_ssh_command(cmd => '! sudo zypper lr', timeout => 600);
```

Comment:
```suggestion
    $instance->run_ssh_command(cmd => 'sudo zypper lr | wc -l', timeout => 600);
```

and then checking for amount of lines is safer approach . in case of current implementation (with '!' ) you will also get green bubble in case of any errors from zypper e.g. broken repo configs

Reply:
Done.